### DISCUSSION TITLE
Support for Returning the Response-Body of Unsuccessful (>= 300) Requests

### DISCUSSION THREAD

**Discussion by efiege** [eclipse-edc/Connector#4252] [General]
Currently the `HttpDataSource` won't forward the request-body from the source, if the http status code is not in the [200..300) range.

However also responses with greater status codes may contain valid response bodys, that are relevant for the consumer.

Is it feasible to also return those response bodies?

Here you can see the current implementation:
https://github.com/eclipse-edc/Connector/blob/893296628a643154cb8811038c06e14c3483f739/extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/pipeline/HttpDataSource.java#L63-L88

**Comment by jimmarino** [comment]
It should not forward the body as this may inadvertently expose internal details of a back-end connection.
